### PR TITLE
Misc changes

### PR DESCRIPTION
**Pull request by hardening**
A set of patches touching various pieces:

* mutualize things in printer channel, so that client and server can share some defines
* fully support saveSessionInfo PDU

### CODE REVIEW

**Comment by hardening** [comment]
@freerdp-bot strange that the test build failed, would you try again ?

**Comment by akallabeth** [comment]
LGTM, +1

**Comment by bmiklautz** [comment]
@hardening maybe you could add the section/spec this is specified in. 
How did you test it?

**Comment by hardening** [comment]
@bmiklautz a modified fireRDS and sending the different kind of messages. I have tested with xfreerdp and mstsc and they looked happy with the sent PDU.

File: include/freerdp/session.h [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+ * limitations under the License.
+ */
+#ifndef __FREERDP_SESSION_H__
+#define __FREERDP_SESSION_H__
+
+#include <winpr/wtypes.h>
+
+/* Logon Information Types */
+#define INFO_TYPE_LOGON			0x00000000
+#define INFO_TYPE_LOGON_LONG		0x00000001
```
Comment (bmiklautz) [inline include/freerdp/session.h]: nitpicking but maybe we can align the first line :)

**Comment by bmiklautz** [comment]
@freerdp-bot test.

**Comment by bmiklautz** [comment]
+1

**Pull request merged**